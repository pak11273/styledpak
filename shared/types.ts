import {
  AlignItemsProps,
  AlignSelfProps,
  BorderRadiusProps,
  ButtonStyleProps,
  ColorProps,
  DisplayProps,
  FlexboxProps,
  FontSizeProps,
  GridProps,
  HeightProps,
  LayoutProps,
  LineHeightProps,
  MaxHeightProps,
  MaxWidthProps,
  MinHeightProps,
  MinWidthProps,
  OrderProps,
  OverflowProps,
  SpaceProps,
  TypographyProps,
  WidthProps,
} from "styled-system"

export type SystemProps = AlignSelfProps &
  AlignItemsProps &
  AlignSelfProps &
  ButtonStyleProps &
  BorderRadiusProps &
  ColorProps &
  DisplayProps &
  FlexboxProps &
  FontSizeProps &
  GridProps &
  HeightProps &
  LayoutProps &
  LineHeightProps &
  MaxHeightProps &
  MaxWidthProps &
  MinHeightProps &
  MinWidthProps &
  OrderProps &
  OverflowProps &
  SpaceProps &
  TypographyProps &
  WidthProps
