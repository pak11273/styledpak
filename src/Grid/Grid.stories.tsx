import Grid from "./Grid"
import { Props } from "../../shared/interfaces"
import React from "react"
import styled from "styled-components"

// Generated with util/create-component.js



export default {
  title: "Grid",
}

const Child = styled.div<Props>`
  grid-area: ${props => props.gridArea || null};
`

export const WithBasicLayout = () => (
  <div>pending</div>
  // <Grid
  //   gridTemplateAreas={["navbar navbar", "sidebar body", "footer footer"]}
  //   gridTemplateColumns="1fr 2fr"
  // >
  //   <Child
  //     style={{
  //       width: "100%",
  //       background: "green",
  //       color: "white",
  //       padding: "5rem",
  //     }}
  //     gridArea="sidebar"
  //   >
  //     sidebar
  //   </Child>
  //   <Child
  //     style={{
  //       width: "100%",
  //       background: "red",
  //       color: "white",
  //       padding: "5rem",
  //     }}
  //     gridArea="navbar"
  //   >
  //     navbar
  //   </Child>
  //   <Child
  //     style={{
  //       width: "100%",
  //       background: "papayawhip",
  //       color: "black",
  //       padding: "5rem",
  //       gridArea: "body",
  //     }}
  //     gridArea="body"
  //   >
  //     body
  //   </Child>
  //   <Child
  //     style={{
  //       width: "100%",
  //       background: "blue",
  //       color: "white",
  //       padding: "5rem",
  //     }}
  //     gridArea="footer"
  //   >
  //     footer
  //   </Child>
  // </Grid>
)
