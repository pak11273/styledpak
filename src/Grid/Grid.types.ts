// Generated with util/create-component.js
export interface GridProps {
  //   foo: string
  gridTemplateAreas?: Array<string>
}
