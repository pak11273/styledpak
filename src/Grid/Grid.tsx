import { GridProps } from "./Grid.types"
import { Props } from "../../shared/interfaces"
import React from "react"
import styled from "styled-components"

const Grid = styled.div`
  display: grid;

  background: ${(props: Props) => props.background || "transparent"};
  grid-template-areas: ${props =>
    props.gridTemplateAreas
      ? props.gridTemplateAreas.reduce(
          (acc, v) => (acc += `${"'" + v + "'" + "\n"}`),
          ``
        ) + ";"
      : "none"};
  grid-template-columns: ${props => props.gridTemplateColumns || "auto"};
  grid-template-rows: ${props => props.gridTemplateRows || "auto"};
  width: ${props => props.width || "100%"};
  max-width: ${props => props.maxWidth || "none"};
`

// const Grid: React.FC<GridProps> = props => {
//   console.log(props)
//   return <Foo {...props}>{props.children}</Foo>

export default Grid
