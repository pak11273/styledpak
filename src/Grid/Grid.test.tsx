import "@testing-library/jest-dom"

import { cleanup, render } from "@testing-library/react"

import Grid from "./Grid"
import { GridProps } from "./Grid.types"
import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"

// Generated with util/create-component.js

describe("Test Component", () => {
  let props: GridProps

  // beforeEach(() => {
  //   props = {
  //     foo: "bar",
  //   }
  // })

  afterEach(cleanup)

  // const renderComponent = () => render(<Grid {...props} />)

  // it("matches snapshot", () => {
  // const tree = renderer.create(<Grid />).toJSON()
  // expect(tree).toMatchSnapshot()
  // })

  // Check props
  it("should have a black background if given a background props of black", () => {
    // const tree = renderer.create(<Grid />).toJSON()
  })

  // nest a component and the label of that component should
  it("renders a child component with the correct areas", () => {
    // const grid = renderer.create(
    // <Grid>
    //   <div style={{ gridArea: "navbar" }}>navbar</div>
    // </Grid>
    // )
  })

  // the grid should be responsive
})
