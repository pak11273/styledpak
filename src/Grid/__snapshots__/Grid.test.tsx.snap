// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component matches snapshot 1`] = `
<div
  className="sc-AxjAm bXOCfU"
  data-testid="grid"
/>
`;
